src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -20,6 +36,15 @@ class ConfigurationMerger {
         return merged
     }
 
+    @NonCPS
+    def static mergeDeepStructure(Map parameters, Map parameterKeys, Map configuration, Map configurationKeys, Map defaults=[:]){
```

Comment:
IMO we should not expose the deep-merge feature via method name. This is confusing. Should be simply ```merge``` instead of ```mergeDeepStructure```. With this it is a sibling of  ```merge(Map parameters, List parametersKeys, Map configurationMap, List configurationKeys, Map defaults)``` (where the keys are provided as List, instead of a Set).

Reply:
deep merge will be handled in #66